src/meta/src/rpc/service/ddl_service.rs

```diff
@@ -307,14 +308,29 @@ where
                     .map_err(tonic_err)?;
                 return Err(e.into());
             }
-            Ok(mut inner_internal_tables) => {
+            Ok(()) => {
                 self.set_table_mapping(&mut mview).map_err(tonic_err)?;
-                for inner_table in &mut inner_internal_tables {
+                let mut internal_table = ctx
+                    .internal_table_id_map
+                    .iter()
+                    .filter(|(_, table)| table.is_some())
+                    .map(|(_, table)| table.clone().unwrap())
+                    .collect_vec();
+
+                for inner_table in &mut internal_table {
                     self.set_table_mapping(inner_table).map_err(tonic_err)?;
                 }
-                inner_internal_tables
+                internal_table
             }
         };
+
+        // tracing for checking the diff of catalog::Table and internal_table_id count
+        tracing::info!(
+            "create_materialized_view internal_table_count {} internal_table_id_count {}",
+            internal_tables.len(),
+            ctx.internal_table_id_map.len()
+        );
```

Comment:
What's the meaning of the diff?

Reply:
internal_table_id count and internal_tables (catalog) may different , because some stream_node (like Lookup, TopN , AppendOnlyTopN) has not own table_catalog now. So add a tracing_info for this case